pkg/types/baremetal/validation/platform.go

```diff
@@ -311,6 +323,8 @@ func ValidatePlatform(p *baremetal.Platform, n *types.Networking, fldPath *field
 
 	allErrs = append(allErrs, validateHostsWithoutBMC(p.Hosts, fldPath)...)
 
+	allErrs = append(allErrs, validateBootMode(p.Hosts, fldPath.Child("Hosts"))...)
```

Comment:
We already have a validateHosts function, https://github.com/openshift/installer/blob/master/pkg/types/baremetal/validation/platform.go#L153

The reason that this belongs there instead of here, is that tells you *which* host had the incorrect value.  Instead of putting the error on the top-level hosts object, you should get one like this:

`baremetal.hosts[0].BootMode: Invalid value: "foobar" must be one of "UEFI" or "legacy"`

Reply:
Once moved we'll need a test as well

Reply:
Tests added